### ISSUE TITLE
Small file-changes does not trigger the cache breaker

### ISSUE THREAD

**Issue by KPeschke** [mrimann/CacheBreaker#2]
It seems that the shortened md5 is not enough to "see" small changes.

**Steps to reproduce:**
1. Generate an uri with cachbreaker for the js file in [v1_Compressed.zip](https://github.com/mrimann/CacheBreaker/files/1311349/v1_Compressed.zip)
2. Remember the cachebreaker-hash (it is "371403e1")
3. Change the file with the js file in [v2_Compressed.zip](https://github.com/mrimann/CacheBreaker/files/1311357/v2_Compressed.zip)

**Expected result:**
New cachebreaker-hash

**Actual result:**
Hash did not change

**Idea/suggestion**: use another method to shorten the md5 (not substr) or take all 32 characters. Maybe put the length into settings so every project can decide how long the hash shall be.

**Comment by mrimann** [comment]
I just checked this, and fear that there must be some different reason for this wrong behaviour. For testing, I downloaded and unpacked your two files into a directory, then created a php-file with the following content:

```
<?php

$filePathA = 'v1_Compressed.js';
$filePathB = 'v2_Compressed.js';

echo substr(md5_file($filePathA), 0, 8);
echo PHP_EOL;
echo substr(md5_file($filePathB), 0, 8);
echo PHP_EOL;

echo ('...');
```

The 2 lines of code with the `substr(md5_file())` call in it, are 1:1 from the code of the CacheBreaker and taken out to test in an isolated area.

Running this php code, I expected to get 2 lines of output, containing two different shortened md5 hashes. But what I got was this:

```
$ php test_md5.php
3cf7691a
ccbc0412
...
```

So I guess the md5 of the whole file differs on "all" of the characters, not only of some of them, so that using `substr()` on it will still detect the different.

This result proves that the code is not completely wrong. But that doesn't help you any further. So here are some things that could lead to the wrong behaviour you've seen:
- file-system caching (it might take a second or two, until an uploaded file is re-loaded and the new version of the file get's hashed)
- NFS mounts (make file-system caching even slower)
- Caching within the Flow- or Neos-Application (could lead to situations where the cache-breaker code itself doesn't get executed at all, thus a cached result is returned)
